src/lib/swapper/swappers/CowSwapper/CowSwapper.ts

```diff
@@ -55,41 +70,33 @@ export class CowSwapper implements Swapper<KnownChainIds.EthereumMainnet> {
     return getCowSwapTradeQuote(this.deps, input)
   }
 
-  executeTrade(
-    args: ExecuteTradeInput<KnownChainIds.EthereumMainnet>,
-  ): Promise<Result<TradeResult, SwapErrorRight>> {
-    return cowExecuteTrade(this.deps, args)
+  executeTrade(args: CowswapExecuteTradeInput<T>): Promise<Result<TradeResult, SwapErrorRight>> {
+    return cowExecuteTrade<T>(this.deps, args)
   }
 
   filterBuyAssetsBySellAssetId(args: BuyAssetBySellIdInput): AssetId[] {
     const { assetIds = [], sellAssetId } = args
     const assets = selectAssets(store.getState())
     const sellAsset = assets[sellAssetId]
 
-    if (
-      sellAsset === undefined ||
-      sellAsset.chainId !== KnownChainIds.EthereumMainnet ||
-      sellAssetId === ethAssetId || // can sell erc20 only
-      COWSWAP_UNSUPPORTED_ASSETS.includes(sellAssetId)
-    )
-      return []
```

Comment:
We still probably want to keep this as early return, to avoid further iterations l.84 if the sell asset is invalid

Reply:
It's added down below